src/osd/OSD.cc

```diff
@@ -3942,7 +3943,34 @@ bool OSD::heartbeat_reset(Connection *con)
   return true;
 }
 
+bool OSD::network_check()
```

Comment:
This name doesn't work; it's doing a lot more than just checking the network. :) Perhaps "network_rebind_and_check"? Or maybe you actually want two functions; one to rebind and one to check the state of the rebound links.